evaluation/results/opt/concatenate_all_results.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import json
+import re
+from pathlib import Path
+from re import Pattern
+from typing import List, Dict
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--results-dir", required=True, type=Path, help="Path to the list of results")
+    parser.add_argument("--concatenate-output-file", required=True, type=Path, help="Path to store the final output file")
+    return parser.parse_args()
+
+MODEL = "opt-175b-meg-ds"
+# MODEL = "global_step95000"
+RESULTS_REGEX = re.compile(rf"(eai|bs)_results_lm-eval_{MODEL}_(\d{4}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2})_backup\.json")
+def get_all_files_that_match_results_in_folder(root_folder: Path, regex: Pattern) -> List[Path]:
+    json_files = []
+    for folder in root_folder.iterdir():
+        if folder.is_dir():
+            json_files += get_all_files_that_match_results_in_folder(folder)
+        else:
+            # it's actually a file
+            file = folder
+
+            match = RESULTS_REGEX.match(file.name)
+
+            if match is None:
+                continue
+            else:
+                json_files.append(file)
+    return json_files
+
+def sort_dict(dictionary: Dict) -> Dict:
+    results = {}
+
+    for key, value in sorted(dictionary.items(), key=lambda item: item[1]):
```

Comment:
```suggestion
    for key, value in sorted(dictionary.items()):
```

Getting 

```python
Traceback (most recent call last):
  File "concat.py", line 102, in <module>
    main()
  File "concat.py", line 93, in main
    final_result = sort_dict(final_result)
  File "concat.py", line 40, in sort_dict
    for key, value in sorted(dictionary.items(), key=lambda item: item[1]):
TypeError: '<' not supported between instances of 'dict' and 'dict'
```

otherwise

Reply:
Hum I think this is what we want.
```suggestion
    for key, value in sorted(dictionary.items(), key=lambda item: item[0]):
```

Reply:
Both work fine 👻